source/adapter/put.js

```diff
@@ -11,13 +11,14 @@ module.exports = {
             .then(responseHandlers.handleResponseCode);
     },
 
-    putFileContents: function putFileContents(url, filePath, data) {
+    putFileContents: function putFileContents(url, filePath, data, options) {
+        //Set default headers
+        options.headers["Content-Type"] = "application/octet-stream";
```

Comment:
What if `options.headers` doesn't exist? This would throw an exception. The options object, if used, should be standardised and robust - a single helper function could be used (private) to normalise the incoming options from the user.

Also, these lines make `Content-Type` and `Content-Length` immutable, as you override them even if the user provides them. They should be optional, for instance: `options.headers["Content-Length"] = options.headers["Content-Length"] || data.length;`.